hoot-core/src/main/cpp/hoot/core/io/OsmApiChangeset.cpp

```diff
@@ -235,6 +234,52 @@ void XmlChangeset::updateChangeset(const QString &changes)
   }
 }
 
+bool XmlChangeset::fixChangeset(const QString& update)
+{
+  /* <osm>
+   *  <node|way|relation id="#" lat="#" lon="#" version="#" changeset="#" user="#" uid="#" visible="#" timestamp="#">
+   *   <tag k="..." v="..."/>
+   *   ...
+   *  </node>
+   * </osm>
+   */
+  bool success = false;
+  QXmlStreamReader reader(update);
+  //  Make sure that the XML provided starts with the <osm> tag
+  QXmlStreamReader::TokenType type = reader.readNext();
+  if (type == QXmlStreamReader::StartDocument)
+    type = reader.readNext();
+  if (type == QXmlStreamReader::StartElement && reader.name() != "osm")
+  {
+    LOG_WARN("Unknown element fix format.");
+    return success;
+  }
+  //  Iterate all of updates and record them
+  while (!reader.atEnd() && !reader.hasError())
+  {
+    type = reader.readNext();
+    if (type == QXmlStreamReader::StartElement)
+    {
+      QStringRef name = reader.name();
+      QXmlStreamAttributes attributes = reader.attributes();
+      long id = 0;
+      long version = 0;
+      if (attributes.hasAttribute("id"))
+        id = attributes.value("id").toString().toLong();
+      if (attributes.hasAttribute("version"))
+        version = attributes.value("version").toString().toLong();
+      //  Fix the element
+      if (name == "node")
+        success |= fixElement(_nodes, id, version);
+      else if (name == "way")
+        success |= fixElement(_ways, id, version);
+      else if (name == "relation")
+        success |= fixElement(_relations, id, version);
+    }
+  }
+  return success;
+}
+
```

Comment:
Parse the OSM XML data that is retrieved from querying the OSM API for a node/way/relation and fix the element's version.